File: client-app/src/app/model/produk.ts
export class Produk {
    public id:String=null;
    public kategori: String=null;
    public kdKategori: String=null;
    public namaProduk: String=null;
    public hargaProduk: number=null;
    public stokProduk: number=null;
    public deskripsiProduk:String=null;
	public fotoProduk1:String=null;
	public fotoProduk2:String=null;
	public fotoProduk3:String=null;
	public fotoProduk4:String=null;
    public fotoProduk5:String=null;
    public jumlahItem:number=null;
}

export class ProdukHome{
    public id:String=null;
    public namaProduk: String=null;
    public hargaProduk: number=null;
    public stokProduk: number=null;
    public fotoProduk1:String=null;
    public image:any=null;
}

export class ProdukHomeDTO{
    public produk:Array<ProdukHome>;
    public totalPage:number;
    public length:number;
}

export class CartDTO{
    public id:number;
    public jumlahItem:number;
    public idUser:number;
    public idProduk:String;
    public namaProduk:String;
    public hargaProduk:number;
    public fotoProduk1:String;
    public urlfotoProduk1:any;
}

export class CartDTOList{
    public id:number;
    public idUser:number;
    public cartDTO:Array<CartDTO>;
}

File: resource-server/src/main/java/com/firstproject/resourceserver/model/entity/Produk.java
package com.firstproject.resourceserver.model.entity;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.Data;

@Data
@Entity
@Table(name = "dat_produk")
public class Produk {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private String id;

	@ManyToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "kategori_produk")
	private KategoriProduk kategoriProduk;

	@Column(name = "nama_produk")
	private String namaProduk;

	@Column(name = "harga_produk")
	private Long hargaProduk;

	@Column(name = "stok_produk")
	private Integer stokProduk;

	@Column(name = "deskripsi_produk")
	private String deskripsiProduk;

	@Column(name = "foto_produk1")
	private String fotoProduk1;

	@Column(name = "foto_produk2")
	private String fotoProduk2;

	@Column(name = "foto_produk3")
	private String fotoProduk3;

	@Column(name = "foto_produk4")
	private String fotoProduk4;

	@Column(name = "foto_produk5")
	private String fotoProduk5;

//	@OneToOne(mappedBy = "produk")
//	private Cart cart;
	
//	@OneToMany(mappedBy = "idProduk", cascade = { CascadeType.ALL })
//	private List<GambarProduk> gambar;

}
